neuralpredictors/layers/readouts/attention.py

```diff
@@ -1,88 +1,96 @@
-        **kwargs,
-    ):
+        in_shape: Tuple[int, int, int],
+        outdims: int,
+        bias: bool,
+        init_noise: float = 1e-3,
+        attention_kernel: int = 1,
+        attention_layers: int = 1,
+        mean_activity: Optional[Mapping[str, float]] = None,
+        feature_reg_weight: float = 1.0,
+        gamma_readout: Optional[float] = None,  # deprecated, use feature_reg_weight instead
+        **kwargs: Any,
+    ) -> None:
         super().__init__()
         self.in_shape = in_shape
         self.outdims = outdims
-        self.feature_reg_weight = self.resolve_deprecated_gamma_readout(feature_reg_weight, gamma_readout)
+        self.feature_reg_weight = self.resolve_deprecated_gamma_readout(feature_reg_weight, gamma_readout)  # type: ignore[no-untyped-call]
         self.mean_activity = mean_activity
         c, w, h = in_shape
         self.features = Parameter(torch.Tensor(self.outdims, c))
 
-        attention = nn.Sequential()
+        attention = nn_modules.Sequential()
         for i in range(attention_layers - 1):
             attention.add_module(
                 f"conv{i}",
-                nn.Conv2d(c, c, attention_kernel, padding=attention_kernel > 1),
+                nn_modules.Conv2d(c, c, attention_kernel, padding=attention_kernel > 1),
             )
-            attention.add_module(f"norm{i}", nn.BatchNorm2d(c))
-            attention.add_module(f"nonlin{i}", nn.ELU())
+            attention.add_module(f"norm{i}", nn_modules.BatchNorm2d(c))  # type: ignore[no-untyped-call]
+            attention.add_module(f"nonlin{i}", nn_modules.ELU())
         else:
             attention.add_module(
                 f"conv{attention_layers}",
-                nn.Conv2d(c, outdims, attention_kernel, padding=attention_kernel > 1),
+                nn_modules.Conv2d(c, outdims, attention_kernel, padding=attention_kernel > 1),
             )
         self.attention = attention
 
         self.init_noise = init_noise
         if bias:
-            bias = Parameter(torch.Tensor(self.outdims))
-            self.register_parameter("bias", bias)
+            bias_param = Parameter(torch.Tensor(self.outdims))
+            self.register_parameter("bias", bias_param)
         else:
             self.register_parameter("bias", None)
         self.initialize(mean_activity)
 
     @staticmethod
-    def init_conv(m):
-        if isinstance(m, nn.Conv2d):
-            nn.init.xavier_normal_(m.weight.data)
+    def init_conv(m: nn_modules.Module) -> None:
+        if isinstance(m, nn_modules.Conv2d):
+            init.xavier_normal_(m.weight.data)
             if m.bias is not None:
                 m.bias.data.fill_(0)
 
-    def initialize_attention(self):
+    def initialize_attention(self) -> None:
         self.apply(self.init_conv)
 
-    def initialize(self, mean_activity=None):
+    def initialize(self, mean_activity: Optional[Mapping[str, float]] = None) -> None:  # type: ignore[override]
```

Comment:
Because the signature of the super type (i.e. `Readout`) is incompatible with the signature of this method. It violates the [Liskov substitution princible](https://en.wikipedia.org/wiki/Liskov_substitution_principle). i.e. code that expects `Readout` could break if you give it an `AttentionReadout` even though it is a subclass of `Readout`.